test/unit/webui/components/search.spec.js

```diff
@@ -0,0 +1,204 @@
+    expect(wrapper.state('loading')).toBeTruthy();
+    expect(wrapper.state('search')).toEqual(newValue);
+  });
+
+  test('handleSearch: cancel all search requests when there is no value in search component with type method', () => {
+    const { handleSearch, requestList } = wrapper.instance();
+    const spy = jest.spyOn(wrapper.instance(), 'cancelAllSearchRequests');
+    const newValue = '';
+
+    handleSearch(event, { newValue, method: 'type' });
+
+    expect(event.stopPropagation).toHaveBeenCalled();
+    expect(wrapper.state('error')).toBeFalsy();
+    expect(wrapper.state('loaded')).toBeFalsy();
+    expect(wrapper.state('loading')).toBeTruthy();
+    expect(wrapper.state('search')).toEqual(newValue);
+    expect(spy).toHaveBeenCalled();
+    expect(requestList).toEqual([]);
+  });
+
+  test('handleSearch: when method is not type method', () => {
+    const { handleSearch } = wrapper.instance();
+    const newValue = '';
+
+    handleSearch(event, { newValue, method: 'click' });
+
+    expect(event.stopPropagation).toHaveBeenCalled();
+    expect(wrapper.state('error')).toBeFalsy();
+    expect(wrapper.state('loaded')).toBeFalsy();
+    expect(wrapper.state('loading')).toBeFalsy();
+    expect(wrapper.state('search')).toEqual(newValue);
+  });
+
+  test('handlePackagesClearRequested: should clear suggestions', () => {
+    const { handlePackagesClearRequested } = wrapper.instance();
+
+    handlePackagesClearRequested();
+
+    expect(wrapper.state('suggestions')).toEqual([]);
+  });
+
+  test('handleFetchPackages: should load the packages from API', async () => {
+    const apiResponse = [{ name: 'verdaccio' }, { name: 'verdaccio-htpasswd' }];
+    const suggestions = [{ label: 'verdaccio' }, { label: 'verdaccio-htpasswd' }];
+
+    jest.resetModules();
+    jest.doMock('../../../../src/webui/utils/api', () => ({
+      request(url) {
+        expect(url).toEqual('search/verdaccio');
+        return Promise.resolve(apiResponse);
+      },
+    }));
+    jest.doMock('lodash/debounce', () => {
+      return function debounceMock(fn, delay) {
+        return fn;
+      };
+    });
+
+    const Search = require(SEARCH_FILE_PATH).default;
+    const component = mount(<Search />);
+    component.setState({ search: 'verdaccio' });
+    const { handleFetchPackages } = component.instance();
+    await handleFetchPackages({ value: 'verdaccio' });
+    expect(component.state('suggestions')).toEqual(suggestions);
+    expect(component.state('error')).toBeFalsy();
+    expect(component.state('loaded')).toBeTruthy();
+    expect(component.state('loading')).toBeFalsy();
+  });
+
+  test('handleFetchPackages: when browser cancel a request', async () => {
+    const apiResponse = {
+      name: 'AbortError',
+    };
+
+    jest.resetModules();
+    jest.doMock('../../../../src/webui/utils/api', () => ({
+      request: jest.fn(() => Promise.reject(apiResponse)),
+    }));
+    jest.doMock('lodash/debounce', () => {
```

Comment:
I'd group within a `describe` all use cases that need this block to avoid duplicates.
```
jest.doMock('lodash/debounce', () => {
      return function debounceMock(fn, delay) {
        return fn;
      };
    });
```